bibliopixel/animation/base.py

```diff
@@ -125,32 +97,15 @@ def run(self, amt=1, fps=None, sleep=None, max_steps=0, until_complete=False,
             max_cycles=0, threaded=False, joinThread=False, callback=None,
             seconds=None):
 
-        self._callback = callback
-        self._led._threadedAnim = threaded
-        if self._led._threadedAnim:
-            self._stopEvent.clear()
-        self._callback = callback
-
         def run():
             try:
                 self._run(amt, fps, sleep, max_steps, until_complete,
                           max_cycles, seconds)
             finally:
                 self.cleanup()
 
-        if self._led._threadedAnim:
-            def target():
-                # TODO: no testpath exercises this code...
-                log.debug('Starting thread...')
-                run()
-                log.debug('Thread Complete')
-
-            self._thread = threading.Thread(target=target, daemon=True)
-            self._thread.start()
-            if joinThread:
-                self._thread.join()
-        else:
-            run()
+        self._callback = callback
```

Comment:
Just an FYI... nothing uses this callback that I'm aware of. It was kind of a good idea at the time but never got used. So this could be added to the list of things to prune if need be.